simpleline/render/io_manager.py

```diff
@@ -175,13 +197,36 @@ def get_user_input_without_check(self, prompt, hidden=False):
 
         See `get_user_input()` method.
         """
-        return self._get_user_input(prompt, hidden)
+        if self._check_prompt(prompt):
+            self._start_user_input_async(prompt, hidden)
+            return self._wait_on_user_input()
+
+    def _check_input_thread_running(self):
+        if self._input_thread is not None and self._input_thread.is_alive():
+            raise KeyError("Can't run multiple input threads at the same time!")
+
+    def _check_prompt(self, prompt):
```

Comment:
Sounds better to me.

I don't know what is `the other way`, it could be practically anything, it even could be generated input which is not from a user. I'm not able to specify anything here it's completely out of my control.